### ISSUE TITLE
PA REVIEW: Tasks

### ISSUE THREAD

**Issue by walterhwan** [walterhwan/remember_the_bacon#5]
- [x] User can create/edit/delete task
- [x] Adequate styling
- [ ] Smooth, bug-free navigation
- [ ] Adequate and appropriate seeds to demonstrate the feature

**Comment by AAlfarho** [comment]
Amazing work on the Task feature, tasks are 1 of the 2 core features of your site, kudos on the attention to detail put in place, kudos!

There are a couple of details we need to address before we can close this MVP.

- [x] Clicking on the `[...]` drop-down shows a `Delete tasks` even if there are no tasks selected, we should disable the `delete task` action unless there is a task selected.
    + ![screen shot 2018-04-11 at 9 52 48 am](https://user-images.githubusercontent.com/1103663/38631275-25578338-3d6e-11e8-86a8-d45319a900e7.png)
- [x] There is only one way of closing the _delete task dropdown_, and that is by clicking on the **...** again, we should be able to close it by clicking everywhere else, otherwise, even if we navigate to a different list/add list, the drop-down will still be visible.
- [x] Changing a task's estimate or to `completed`, is not updating the overall `List` information, only after a refresh, this information is in sync.
- [x] Currently, the only way to update the _value_ of tasks it's by clicking it, let's add something like _rememberthemilk_ to let users know they can click the task name to update its value. 
    ![screen shot 2018-04-11 at 9 43 09 am](https://user-images.githubusercontent.com/1103663/38630802-c928ced8-3d6c-11e8-9abe-2d28b6eb486f.png)
- [x] The only way to persist a new value for a task its by hitting `enter` after updating the value. This means that changes will not be saved if users click on a different task or click the _close button_ without hitting `enter`, this is not intuitive, we should  **autosave** the new task's value on the spot, we can do this several ways.
   + Have a debounced function that saves the task value on input change, **NOTE**, this does not mean trigger a new dispatch and hit our server every time the user updates the input, we should throttle or debounce this functionality, take a look at _lodash_ implementation (https://css-tricks.com/debouncing-throttling-explained-examples/).
  + On componentWillReceiveProps or on the close action, we can trigger an update, I personally like the previous option better, given that your other components (Tasks Index) will update live.
- [x] We should remove the `add task input` when switching to the `completed` tab.


Assign this issue back to me once the updates addressing the feedback is live on Heroku.

**Comment by AAlfarho** [comment]
Kudos, this MVP is now complete.
